### PR TITLE
cleanup orphaned hosted zones from control plane

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/4212.

### CODE REVIEW

File: pkg/cleaner/aws/aws.go [hunk 1]
```diff
@@ -126,6 +135,40 @@ func (a *Cleaner) cleanBuckets() error {
...
+		for _, hz := range out.HostedZones {
+			if hz.Name == nil || hz.Id == nil {
+				continue
+			}
+
+			fmt.Printf("\n")
+			fmt.Printf("%#v\n", *hz.Id)
+			fmt.Printf("%#v\n", *hz.Name)
+			fmt.Printf("%#v\n", hz)
+			fmt.Printf("\n")
```
Comment (xh3b4sd) [inline pkg/cleaner/aws/aws.go]: This is just some listing of hosted zones. I do not know how to identify the ones we can just remove. I need feedback from @giantswarm/sig-infrastructure. <3

File: pkg/cleaner/aws/spec.go [hunk 2]
```diff
@@ -21,6 +22,10 @@ type CFClient interface {
 	UpdateTerminationProtection(*cloudformation.UpdateTerminationProtectionInput) (*cloudformation.UpdateTerminationProtectionOutput, error)
 }
 
+type Route53Client interface {
+	ListHostedZones(input *route53.ListHostedZonesInput) (*route53.ListHostedZonesOutput, error)
+}
```
Comment (kopiczko) [inline pkg/cleaner/aws/spec.go]: We do we need this interface?

Comment (xh3b4sd) [reply pkg/cleaner/aws/spec.go]: No idea. These have been there before for the other clients so I just aligned my changes with it.

**Comment by calvix** [comment]
We could either:
* parse e2e id from the zone and then check for some other resource (like ec2 instances) and if there are no instances or intances created 24h hours ago we could delete it
* add some timestamp info a description of host zone which would give cleaner info if it can be removed or not

**Comment by xh3b4sd** [comment]
I try going with the EC2 instance magic. Thanks.

**Comment by xh3b4sd** [comment]
@giantswarm/sig-infrastructure I checked again and parsing the tenant ID from the hosted zone name does not work because the hash in there is not a tenant ID. I do not know how this works and what this is. Would be cool to talk.

**Comment by kopiczko** [comment]
> parsing the tenant ID

I think those are control plane tests leftovers.

**Comment by calvix** [comment]
So after a chat with Tim and little investigation: 
the id of installation from `e2e-terraform-c5f600.eu-central-1.aws.gigantic.io.` is  `e2e-terraform-c5f600`.
This is not tenant cluster so the methods for tenant cluster might not work. all components in AWS should be prefixed with this ID.

the resources starting with `e2e-cluster-xxxx` were from the older version of terraform where naming was different. I cleaned them manually for now.

IF you want to have some json data in the comment in order to better identify the hosted zone then please post a format and I can do PR to terraform.

**Comment by calvix** [comment]
I would then add this to comments for now:
{"last_updated": "2018-09-28T15:00:00Z", "managed_by":"terraform"}

Possibly any resource could be left over. The cleanup is done by terraform and it can fail at some bug or special case.

**Comment by xh3b4sd** [comment]
I would say the PM is then addressed by adding the JSON blob to the comments and merging the ground work we have here. I will just remove the logging. Later we can iterate further. I just want to get this done.

**Comment by xh3b4sd** [comment]
So there is some structure now but I disabled the task because it is not doing anything. We can make use of it later though in case we have it so this here helps addressing the PM a bit.

**Comment by xh3b4sd** [comment]
Fishing for LGTM.

**Pull request merged**